packages/scene-composer/src/components/three-fiber/EntityGroup/index.tsx

```diff
@@ -111,7 +115,7 @@ const EntityGroup = ({ node }: IEntityGroupProps) => {
         dispose={null}
         onPointerDown={onPointerDown}
         onPointerUp={onPointerUp}
-        userData={{ nodeRef: !isEnvironmentNode(node) ? nodeRef : undefined, componentTypes }} // Do not add ref for environment nodes
+        userData={{ nodeRef: !isEnvironmentNode(node) ? nodeRef : undefined, targetRef, componentTypes }} // Do not add ref for environment nodes
```

Comment:
What is the "targetRef"?

Reply:
Since submodels are separate scenes we need to track where the original ref was (hence `targetRef`). When we look at the parent of a submodel its parent is the root of its separate scene & that was what caused the odd placement of the tags when clicking on submodels. By passing an optional `targetRef`, we're able to accurately tell the tag where to be placed in the world.